src/main/java/org/gridsuite/modification/server/modifications/NetworkModificationApplicator.java

```diff
@@ -122,4 +128,33 @@ private NetworkModificationException handleException(NetworkModificationExceptio
             .build());
         return networkModificationException;
     }
+
+    public static TypedValue computeHighestSeverityFromReporterModel(ReporterModel reporter) {
+        Stream<TypedValue> reportsSeverities = reporter.getReports().stream()
+                .map(NetworkModificationApplicator::computeHighestSeverityFromReport);
+
+        Stream<TypedValue> subReportersSeverities = reporter.getSubReporters().stream()
+                .map(NetworkModificationApplicator::computeHighestSeverityFromReporterModel);
+
+        return Stream.concat(reportsSeverities, subReportersSeverities)
+                .max(new TypedValueComparator()).orElse(TypedValue.TRACE_SEVERITY); // orElse not necessary but we never know
+    }
+
+    public static TypedValue computeHighestSeverityFromReport(Report report) {
+        return report.getValues().entrySet().stream()
+                .filter(entry -> entry.getKey().equals(Report.REPORT_SEVERITY_KEY))
+                .map(Map.Entry::getValue)
+                .reduce((currentValue, nextValue) -> Collections.max(List.of(currentValue, nextValue), new TypedValueComparator()))
+                .orElse(TypedValue.TRACE_SEVERITY);
+    }
+
+    public static NetworkModificationResult.ApplicationStatus convertTypedValueToApplicationStatus(TypedValue typedValue) {
+        if (typedValue.equals(TypedValue.ERROR_SEVERITY)) {
```

Comment:
A switch could be used

Reply:
Nope because it's not the same type, you can check